Source/Adapters/Blackjack.Adapters/UserAdapter.cs

```diff
@@ -54,6 +62,19 @@ public async Task<UserBE> GetUsers(string username)
         return usersBySoundex?.SingleOrDefault(user => user.Username.Equals(username, StringComparison.OrdinalIgnoreCase));
     }
 
+    public async Task<UserBE> UpdateUser(UserBE user)
+    {
+        UserBE dbUser = await GetUsers(user.Username);
```

Comment:
If you have the Id of a chunk of data never search on anything else.

Make a new way to fetch Users by UserID if you don't have it already and use that here instead.

Reply:
So I just realized that this is doing a fuzzy search on the soundex instead of an exact search, for practice. I remember you saying something similar earlier as well and that you don't really want the front end to be carrying around anything other than the ID. So in a real world scenario this method should only ever be used within the confines of the backend, in our case probably just the CreateUser method to see if an actual username exists and/or for logging in? 

Otherwise the UpdateUser should have its own GetUserByID because we have that exact bit of information and searching on the PK is fast as well?

Reply:
GetUser via username is for when you need to get users via a username, e.g. logging in and no ID is known yet.

Performing an update against an explicit user you know the ID of you should never do a fuzzy search to hope you happen to grab the right row and happen to update the right row.

You use fuzzy search when it's the right tool for the job.